onmetal_v1_smart.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+#
+# Script for monitoring remaining useful lifetime of OnMetal v1 SATADOM.
+#
+# Requires the following binaries installed & on path:
+#  - smartctl
+#  - lsblk
+#
+# Suggested alarm criteria:
+#
+# if (metric['percent_pe_cycles_used'] >= 1) {
+#     return new AlarmStatus(CRITICAL, 'Drive is beyond expected life.')
+# }
+#
+# if (metric['percent_pe_cycles_used'] >= .8) {
+#     return new AlarmStatus(WARNING, 'Drive >= 80% of its expected life.')
+# }
+#
+# return new AlarmStatus(OK, 'Drive less than 80% through its expected life.')
+
+import subprocess
+
+DEVICE = "/dev/sda"
+
+SATADOM_PE_MAX = {
+    '32G MLC SATADOM': 3000,
+    '7 PIN  SATA FDM': 3000,
+    'Fastable SD 131 7': 3000,
+    'SATADOM-SH TYPE C 3SE': 100000,
+}
+
+
+def _fail(msg="Unknown Error"):
+    print("status err {}".format(msg))
+
+
+def _get_smartctl_attributes():
+    try:
+        out = subprocess.check_output(['smartctl', '--attributes', DEVICE])
+    except:
+        _fail("failed running smartctl")
```

Comment:
I would re-raise the exception here, or move the try/except block outside of this function and put it around its call-site instead.  See comment below.